pkg/archive/archive.go

```diff
@@ -382,6 +382,56 @@ func createTarFile(path, extractDir string, hdr *tar.Header, reader io.Reader, L
 	return nil
 }
 
+// Matches returns true if relFilePath matches any of the patterns
+// and isn't excluded by any of the subsequent patterns.
```

Comment:
Let's add something like:
```
// This func is basically the same as fileutils.Match() but optimized for archive.go.
// This will assume that the inputs have been preprocessed and therefore we 
// don't need to do as much error checking/clean-up. This was done to avoid
// repeating these steps on each file being checked during the archive process.
// The more generic fileutils.Match() can't make these assumptions.
```
so that people know why we duplicated it.
Perhaps even add a comment to the fileutils's one pointing to here so that if
people change one they know to change the other.